meson.build

```diff
@@ -0,0 +1,54 @@
+# This file is adapted from https://github.com/scipy/scipy/blob/main/meson.build
+project(
+    'pandas',
+    'c', 'cpp', 'cython',
+    version: '2.0.0.dev0',
+    license: 'BSD-3',
+    meson_version: '>=0.64',
+    default_options: [
+        # TODO: investigate, does meson try to compile against debug Python
+        # when buildtype = debug, this seems to be causing problems on CI
+        # where provided Python is not compiled in debug mode
+        'buildtype=release',
+        # TODO: turn on werror when ready
+        #'werror=true',
+        # TODO: We are using POSIX functions(strdup in ujson), so we can't compile
```

Comment:
I think the issue here is that in setuptools we define `_GNU_SOURCE` but that didn't carry over to this implementation. If you remove `-D_GNU_SOURCE` on main from setup.py you will also get build warnings for the json module

Reply:
I think #49780 will fix this